llarp/dns_rectypes.cpp

```diff
@@ -0,0 +1,175 @@
+#include <llarp/dns_rectypes.hpp>
+#include <llarp/dns.hpp>  // for vput16bits()
+
+namespace llarp
+{
+  namespace dns
+  {
+    bool
+    record::parse(std::vector< byte_t > bytes)
+    {
+      return bytes.size() ? true : false;
+    };
+
+    std::vector< byte_t >
+    record::to_bytes()
+    {
+      std::vector< byte_t > retval;
+      return retval;
+    };
+
+    type_1a::type_1a() : record()
+    {
+      this->ipaddr.h = 0;
+    }
+
+    bool
+    type_1a::parse(std::vector< byte_t > bytes)
+    {
+      if(bytes.size() < 4)
+      {
+        LogWarn("Less than 4 bytes passed in");
+        return false;
+      }
+      // endian problems? no, it should come in, in network order
+      /*
+      LogDebug("Read ", std::to_string(bytes[0]), ".",
+              std::to_string(bytes[1]), ".",
+              std::to_string(bytes[2]), ".",
+              std::to_string(bytes[3]));
+      */
+      this->ipaddr = ipaddr_ipv4_bits(bytes[3], bytes[2], bytes[1], bytes[0]);
+      // LogDebug("Test ", this->ipaddr);
+      return bytes.size() ? true : false;
+    };
+
+    std::vector< byte_t >
+    type_1a::to_bytes()
+    {
+      std::vector< byte_t > retval;
+      vput16bits(retval, 4);               // rdLength
+      vput32bits(retval, this->ipaddr.h);  // write IP
+      return retval;
+    };
+
+    type_2ns::type_2ns() : record()
+    {
+      this->ns = "";
+    }
+    bool
+    type_2ns::parse(std::vector< byte_t > bytes)
+    {
+      // trim last 2 bytes... probably the size
+      this->ns = std::string((char *)bytes.data(), bytes.size() - 2);
```

Comment:
`static_cast` / `reinterpret_cast` where applicable.

Reply:
had to use reinterpret_cast, it didn't like static_cast